buildpack.go

```diff
@@ -0,0 +1,94 @@
+package libdependency
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/BurntSushi/toml"
+	"github.com/joshuatcasey/collections"
+	"github.com/joshuatcasey/libdependency/versionology"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+)
+
+// ParseBuildpackToml takes in a path to a buildpack.toml and parses that into a cargo.Config
+func ParseBuildpackToml(buildpackTomlPath string) (cargo.Config, error) {
+	if config, err := cargo.NewBuildpackParser().Parse(buildpackTomlPath); err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return cargo.Config{}, fmt.Errorf("unable to open buildpack.toml: %w", err)
+		} else if tomlError, ok := err.(toml.ParseError); ok {
+			return cargo.Config{}, fmt.Errorf("unable to parse buildpack.toml: %s", tomlError.ErrorWithPosition())
+		} else { //untested
+			return cargo.Config{}, err
+		}
+	} else {
+		return config, nil
+	}
+}
+
+// GetDependenciesById will return an array of dependencies with the given id, of a type that "extends"
+// cargo.ConfigMetadataDependency and implements versionology.VersionFetcher
```

Comment:
Should this be:
```suggestion
// cargo.ConfigMetadataDependency and implements versionology.Dependency
```

Reply:
`versionology.Dependency` implements `versionology.VersionFetcher` so the overall statement is "correct" if not intuitive:

> // GetDependenciesById will return an array of dependencies with the given id, of a type that "extends"
// cargo.ConfigMetadataDependency and implements versionology.VersionFetcher